src/utils/guildRole.ts

```diff
@@ -135,3 +142,70 @@ export async function getGuildRoleByName(
   const roles = await getGuildRoles(env);
   return roles?.find((role) => role.name === roleName);
 }
+
+export async function mentionEachUserInMessage({
+  message,
+  userIds,
+  channelId,
+  env,
+}: {
+  message?: string;
+  userIds: string[];
+  channelId: number;
+  env: env;
+}) {
+  const batchSize = 5;
+  let waitTillNextAPICall = 0;
+  try {
+    const failedUsers: Array<string> = [];
+    for (let i = 0; i < userIds.length; i += batchSize) {
+      const batchwiseUserIds = userIds.slice(i, i + batchSize);
+      const messageRequest = batchwiseUserIds.map((userId) => {
+        return fetch(`${DISCORD_BASE_URL}/channels/${channelId}/messages`, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bot ${env.DISCORD_TOKEN}`,
+          },
+          body: JSON.stringify({
+            content: `${message ? message + " " : ""} ${userId}`,
+          }),
+        }).then((response) => {
+          const rateLimitRemaining = parseRateLimitRemaining(response);
+          if (rateLimitRemaining === 0) {
+            waitTillNextAPICall = Math.max(
+              parseResetAfter(response),
+              waitTillNextAPICall
+            );
+          }
+          return response.json();
+        }) as Promise<discordMessageRequest | discordMessageError>;
+      });
+      const responses = await Promise.all(messageRequest);
```

Comment:
If one call in batch fails then?

Reply:
We ensure this by looping through the responses to collect information about the API call. From there, we store the users from the failed API calls into an array. Finally, we send a message to inform them that these users could be tagged in individual messages.